### PR TITLE
Fix up the is_complete issue

### LINKED ISSUE #2998: Double complex errors
```
double complexes: Test Failed at /home/runner/work/Oscar.jl/Oscar.jl/experimental/DoubleComplex/test/DoubleComplex.jl:138
  Expression: !(is_complete(KK))

Stacktrace:
 [1] macro expansion
   @ /opt/hostedtoolcache/julia/1.9.3/x64/share/julia/stdlib/v1.9/Test/src/Test.jl:478 [inlined]
 [2] macro expansion
   @ ~/work/Oscar.jl/Oscar.jl/experimental/DoubleComplex/test/DoubleComplex.jl:138 [inlined]
 [3] macro expansion
   @ /opt/hostedtoolcache/julia/1.9.3/x64/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
 [4] top-level scope
   @ ~/work/Oscar.jl/Oscar.jl/experimental/DoubleComplex/test/DoubleComplex.jl:2
double complexes: Test Failed at /home/runner/work/Oscar.jl/Oscar.jl/experimental/DoubleComplex/test/DoubleComplex.jl:141
  Expression: !(is_complete(KK))
```
As seen in https://github.com/oscar-system/Oscar.jl/actions/runs/6777108961/job/18420060848?pr=2996#step:7:2641.

@HechtiDerLachs

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
This should resolve #2998 .

@benlorenz : Thanks a lot for catching this! I would probably  have spent days to find out about what the problem actually is.

### CODE REVIEW

**Comment by benlorenz** [comment]
While it doesn't seem to cause issues right now can you please change the `has_*_bound` stuff as well? These rely on `isdefined` for an `Int64` field which is also always true.

**Comment by HechtiDerLachs** [comment]
True. Thanks again.

File: experimental/DoubleComplex/src/Attributes.jl [hunk 1]
```diff
@@ -20,15 +20,14 @@ function vertical_map_cache(D::DoubleComplexOfMorphisms)
...
+has_right_bound(D::DoubleComplexOfMorphisms) = D.right_bound !== nothing
+has_left_bound(D::DoubleComplexOfMorphisms) = D.left_bound !== nothing
+has_upper_bound(D::DoubleComplexOfMorphisms)   = D.upper_bound !== nothing
+has_lower_bound(D::DoubleComplexOfMorphisms)   = D.lower_bound !== nothing
 
 function is_complete(D::DoubleComplexOfMorphisms)
-  if isdefined(D, :is_complete) && D.is_complete
-    return true
-  end
+  D.is_complete isa Bool && D.is_complete && return true
```
Comment (fingolfin) [inline experimental/DoubleComplex/src/Attributes.jl]: Could also do
```suggestion
  D.is_complete == true && return true
```
or
```suggestion
  D.is_complete === true && return true
```

File: experimental/DoubleComplex/src/Attributes.jl [hunk 2]
```diff
@@ -129,23 +128,23 @@ function vertical_range(D::DoubleComplexOfMorphisms)
 end
 
 function right_bound(D::DoubleComplexOfMorphisms) 
-  isdefined(D, :right_bound) || error("complex has no known horizontal upper bound")
-  return D.right_bound
+  D.right_bound !== nothing || error("complex has no known horizontal upper bound")
+  return D.right_bound::Int
```
Comment (fingolfin) [inline experimental/DoubleComplex/src/Attributes.jl:132]: This annotation shouldn't be necessary, as the compiler can prove using the previous line that `Nothing` is ruled out. That said, it also doesn't hurt, so it's fine to leave it in.

**Review by benlorenz — APPROVED** [review]
Thanks

**Pull request merged**